test/discuss/blog_test.exs

```diff
@@ -11,16 +12,16 @@ defmodule Discuss.BlogTest do
     @valid_attrs %{title: "Some title"}
 
     test "list_topics/0 returns all topics ordered by title" do
-      first_topic = BlogFactory.insert(:topic, %{title: "A new topic"})
-      second_topic = BlogFactory.insert(:topic, %{title: "Second topic"})
+      first_topic = insert(:topic)
+      second_topic = insert(:topic)
 
-      assert Blog.list_topics() == [first_topic, second_topic]
+      assert Blog.list_topics() |> Repo.preload(:user) == [first_topic, second_topic]
```

Comment:
How can it be improved? Instead of using preload, can we use some pattern matching here?

Reply:
Yeah, use pattern matching!